### PR TITLE
sanitize input to write_table

### PR DESCRIPTION
**Pull request by changhiskhan**
otherwise it segfaults if you pass in a non-table

### CODE REVIEW

File: python/lance/tests/test_pytorch.py [hunk 1]
```diff
@@ -47,8 +47,7 @@ def test_data_loader(tmp_path: Path):
     assert torch.equal(value_batch, torch.tensor([10, 11, 12, 13]))
 
 
-@pytest.mark.skipif(platform.system() != "Linux",
-                    reason="there's a bug on mac")
+@pytest.mark.skipif(platform.system() != "Linux", reason="there's a bug on mac")
```
Comment (eddyxu) [inline python/lance/tests/test_pytorch.py:50]: This has already been fixed on mac?

File: python/lance/_lib.pyx [hunk 2]
```diff
@@ -159,6 +159,10 @@ cdef class LanceFileFormat(FileFormat):
 def WriteTable(table: Table,
                sink: Union[str, Path],
                batch_size: int):
+    if not isinstance(table, Table):
```
Comment (eddyxu) [inline python/lance/_lib.pyx:162]: should this function accept the same types as 

https://arrow.apache.org/docs/python/generated/pyarrow.dataset.write_dataset.html ?

File: python/lance/tests/test_pytorch.py (hunk 1 above)
Comment (changhiskhan) [reply python/lance/tests/test_pytorch.py:50]: Weird, I did not add that in. Not sure where it came from ...

Comment (changhiskhan) [reply python/lance/tests/test_pytorch.py:50]: Oh nvm it's just formatting. So it just needs to be removed

File: python/lance/_lib.pyx (hunk 2 above)
Comment (changhiskhan) [reply python/lance/_lib.pyx:162]: It doesn't handle those other types does it?

Comment (eddyxu) [reply python/lance/_lib.pyx:161]: it as `write_dataset`? or `WriteTable`?

Comment (changhiskhan) [reply python/lance/_lib.pyx:161]: this is our WriteTable. We can't customize write_dataset in pyarrow?

**Pull request closed without merging**